### DISCUSSION TITLE
Don't clear error while failed query is refetching

### DISCUSSION THREAD

**Discussion by Tarow** [TanStack/query#6910] [Q&A] [answered]
When a query initially fails, i would like to display the error and keep trying to refetch in the background using `refetchInterval`, until eventually i might get data back.

Now whenever the query is refetching, the previous error is cleared.
An example is here: https://stackblitz.com/edit/tanstack-query-gldaab?file=src%2Findex.jsx

Every five seconds the query refetches and clears the previous error while it's fetching.
Is there a way to persist the `error` and just try to refetch in the background?

**Comment by TkDodo** [comment, marked as answer]
don't think so. Errors are explicitly reset when a fetch is triggered:

https://github.com/TanStack/query/blob/b32ad24f5d2ac79eb8dee32047195e2bb9267b2b/packages/query-core/src/query.ts#L549-L552

**Comment by KODerFunk** [comment]
@Tarow 
Basic concept
```ts
const { error, isSuccess, isFetched } = useQuery(...);
const errorRef = useRef(error);
if (error || isSuccess) errorRef.current = error;

console.log('Preserved error:', { isFetching, error: errorRef.current });
```
